### DISCUSSION TITLE
[2024-01-15] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#8755] [Weekly Demo Day]
- James: Add parser for line by line processing #8719
- Radosław: Add tests for `Enso_Secret`s, update to new cloud API #8736
- Radosław: Remove `set_user_info` from URI #8738
- Radosław: Allow secrets in auth headers #8761 
- Radosław: Type Inference PoC 1 #8652
  - Implemented local binding propagation
  - Vector literal `[x, y, z]` now gets typed as `Vector`.
  - Implemented typing for `case` statements:
     - `case x of 1 -> 1 | 2 -> "a"` is typed as `Integer | Text` 
       - _(the `|` in `case of` is not real syntax, just using to show examples in a single line)_
     - `case x of (i : Integer) -> i | _ -> 0` is typed as `Integer`
   - Ability to check type of function argument if its known (PoC, does not take conversions into account).
   - Next steps: fix serialization, fix a bug (not invokable warning false positive on 'higher order' types in type ascriptions), cleanup, prepare for merging first prototype version. Possibly look into conversions/method resolution, but that's most likely for later.
- Greg: Support on_problems=Problem_Behavior.Report_Warning and Map_Error wrapping in Vector.map #8595
- Greg: Review Nothing in joins, equality and comparator between InMemory and InDB #5156
- Ad: Add new text_left and text_right functions #8691
- Jaroslav: Updating Espresso support & docs to match GraalVM for JDK21 #8641
- Jaroslav: Avoid using cached bindings when the module isn't loaded from cache #8669
- Jaroslav: Chained if_then_else application change #8671
- Jaroslav: Executing (parts of) Truffle TCK with Enso values #8685
- Jaroslav: Replace Jackson library by simple data I/O streams #8693
- Jaroslav: Make suspended atom fields work for boxed atoms #8712
- Hubert: Suppress logs for test cases unless a failure is reported #8694
- Hubert: Optimize AliasAnalysis hotspot #8701
- Hubert: Generate UUIDs on demand #8728
- Hubert: Investigating assertion failure in Greg's PR #8595
- Hubert: Investigating stability issues when applying edits
- Pavel: Refactor Standard.Test to the builder API #8622
- Pavel: language server tests do not unnecessarily re-assemble fat jar #8695
- Pavel: Benchmark downloader script should take the latest score, not the smallest #8707
- Pavel: Fix `Runtime.assert` #8742
- Dmitry: Do not change project module name to Upper_Snake_Case #8697
- Dmitry: Return function schema for atom constructors #8743

**Comment by radeusgd** [comment]
First static type error in Enso:
![image](https://github.com/enso-org/enso/assets/1436948/1579eb27-a6d4-4940-b9cc-e4aef0dc05a1)
yielding
![Zrzut ekranu 2024-01-13 161453](https://github.com/enso-org/enso/assets/1436948/205d10be-cebc-4873-afce-748152710c3c)

However this is still a prototype, because it does not take conversions into account, so it could give false positives if enabled currently.
